src/OSPSuite.Core/Domain/Services/InitialConditionsCreator.cs

```diff
@@ -94,21 +94,31 @@ public InitialCondition CreateInitialCondition(ObjectPath moleculeAmountPath, Mo
          return initialCondition;
       }
 
-      private InitialCondition createInitialConditionForBuildingBlock(IBuildingBlock initialConditionsBuildingBlock, IEntity container, MoleculeBuilder molecule)
+      private InitialCondition createInitialConditionForBuildingBlock(IBuildingBlock initialConditionsBuildingBlock, IContainer container, MoleculeBuilder molecule)
       {
          var initialCondition = createInitialConditionWithValue(container, molecule);
          setInitialConditionFormula(molecule.DefaultStartFormula, initialCondition, formula => _cloneManagerForBuildingBlock.Clone(formula, initialConditionsBuildingBlock.FormulaCache));
          return initialCondition;
       }
 
-      private InitialCondition createInitialConditionWithValue(IEntity container, MoleculeBuilder molecule)
+      private InitialCondition createInitialConditionWithValue(IContainer container, MoleculeBuilder molecule)
       {
-         var initialCondition = CreateInitialCondition(_entityPathResolver.ObjectPathFor(container), molecule.Name, molecule.Dimension, molecule.DisplayUnit);
+         var initialCondition = CreateInitialCondition(objectPathForContainer(container), molecule.Name, molecule.Dimension, molecule.DisplayUnit);
          initialCondition.Value = molecule.GetDefaultInitialCondition();
          return initialCondition;
       }
 
-      private void addMoleculesFrom(InitialConditionsBuildingBlock initialConditionsBuildingBlock, IEntity container, IEnumerable<MoleculeBuilder> molecules)
+      private ObjectPath objectPathForContainer(IContainer container)
+      {
+         var objectPathForInitialCondition = _entityPathResolver.ObjectPathFor(container);
+         var rootContainer = container.RootContainer;
+         if (rootContainer.ParentPath != null)
```

Comment:
This test works, but I'm not sure if I understand the usage of ParentPath fully.

Basically, if you find that the root container has additional path, then this prepends it to the initial condition.